public/app/features/alerting/TestRuleButton.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { PureComponent } from 'react';
+import { JSONFormatter } from 'app/core/components/JSONFormatter/JSONFormatter';
+import { getBackendSrv } from 'app/core/services/backend_srv';
+import { DashboardModel } from '../dashboard/dashboard_model';
+
+export interface Props {
+  panelId: number;
+  dashboard: DashboardModel;
+  LoadingPlaceholder: any;
```

Comment:
Would be awesome if we could create a generic interface for handling `LoadingPlaceholder` prop down the road. Extracting `LoadingPlaceholder` from `AlertTab` component file and placing it in grafana/ui package would be a good starting point.

Also, having prop names in camelCase would be good(vide [Airbnb styleguide](https://github.com/airbnb/javascript/tree/master/react#props))

Reply:
Good suggestion, copied existing LoadingPlaceHolder from QueryInspector. Not sure if this should be part of this PR though.

Reply:
Good point. Created an issue for that #14793